src/crypto/utils.ts

```diff
@@ -1,19 +1,15 @@
 import * as secp from '@noble/secp256k1'
+import {
+  Hex,
+  getAddress,
+  hexToSignature,
+  keccak256,
+  hexToBytes,
+  bytesToHex as viemBytesToHex,
+} from 'viem'
 
 export const bytesToHex = secp.utils.bytesToHex
 
-export function hexToBytes(s: string): Uint8Array {
```

Comment:
Given the sensitivity of some of this stuff, I'm going to suggest we do an extra validation test and try and send some messages using this version of the SDK to an older version of the SDK.

Because the tests mostly use new wallets they can hide some sneaky issues around signature verification.

Reply:
before i start testing client compatibility, i decided to run some tests locally and they passed just fine with a million randomly generated hex values.

```ts
import { equalBytes, oldHexToBytes } from '../../src/crypto/utils'
import { hexToBytes, bytesToHex } from 'viem'
import { randomBytes, webcrypto } from 'node:crypto'

describe('hexToBytes', function () {
  it('old function generates the same as the new function (crypto)', async function () {
    const hexes = Array.from({ length: 1000000 }).map(() =>
      bytesToHex(randomBytes(32))
    )
    hexes.forEach((hex) => {
      expect(equalBytes(oldHexToBytes(hex), hexToBytes(hex)))
    })
  })

  it('old function generates the same as the new function (webcrypto)', async function () {
    const hexes = Array.from({ length: 1000000 }).map(() =>
      bytesToHex(webcrypto.getRandomValues(new Uint8Array(32)))
    )
    hexes.forEach((hex) => {
      expect(equalBytes(oldHexToBytes(hex), hexToBytes(hex)))
    })
  })
})
```

Reply:
testing between different client versions seems to work!

![Screenshot 2024-02-28 at 11 09 20 AM](https://github.com/xmtp/xmtp-js/assets/2925672/14863c2f-07d5-42f2-a25a-5af75c068c08)

Reply:
Appreciate you doing the extra validation here.